core/matrix/sparsity.cpp

```diff
@@ -159,8 +162,12 @@ std::unique_ptr<Sparsity<ValueType, IndexType>>
 Sparsity<ValueType, IndexType>::to_adjacency_matrix() const
 {
```

Comment:
add assert to check if the matrix is square

Reply:
I added it in PR #350 , but I will move it here.